lib/icingadb/icingadb-objects.cpp

```diff
@@ -572,6 +577,39 @@ std::vector<String> IcingaDB::GetTypeOverwriteKeys(const String& type)
 	return std::move(keys);
 }
 
+std::vector<String> IcingaDB::GetTypeDumpSignalKeys(const String& type, bool hasCustomVars)
+{
+	std::vector<String> keys = {
+			m_PrefixConfigObject + type,
+	};
+
+	if (hasCustomVars) {
```

Comment:
Why not just pass the `Type::Ptr` to this function and do the check here? This would also allow avoiding string comparisons below.